base/compiler/inferencestate.jl

```diff
@@ -373,6 +373,31 @@ function InferenceState(result::InferenceResult, cache::Symbol, interp::Abstract
     return InferenceState(result, src, cache, interp)
 end
 
+"""
+    has_non_dt_typevar(typ, v) -> Bool
```

Comment:
I think we have an existing version of this in Test which is much less conservative (it does methods lookups too)? Should we try to unify them?